From 1eb93d9229c54baa5f1b7ccf7d105d3692c51a4d Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Tue, 21 May 2019 16:18:36 +0900
Subject: [PATCH] scd: Fix for SCARD_IO_REQUEST structure.

* scd/apdu.c (struct pcsc_io_request_s): Use pcsc_dword_t for Windows.

--

This fix is for correctness and for the future when we will support
64-bit Windows.

GnuPG-bug-id: 4454
Suggested-by: Juris Ozols
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
---
 scd/apdu.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/scd/apdu.c b/scd/apdu.c
index ffa28c689..254c74101 100644
--- a/scd/apdu.c
+++ b/scd/apdu.c
@@ -256,8 +256,13 @@ static npth_mutex_t reader_table_lock;
 
 struct pcsc_io_request_s
 {
+#if defined(_WIN32) || defined(__CYGWIN__)
+  pcsc_dword_t protocol;
+  pcsc_dword_t pci_len;
+#else
   unsigned long protocol;
   unsigned long pci_len;
+#endif
 };
 
 typedef struct pcsc_io_request_s *pcsc_io_request_t;
-- 
2.32.0.windows.2

